pkg/deployment/ingester.go

```diff
@@ -0,0 +1,159 @@
+package deployment
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/io/v1alpha1"
+	"github.com/jaegertracing/jaeger-operator/pkg/config/sampling"
+	"github.com/jaegertracing/jaeger-operator/pkg/service"
+	"github.com/jaegertracing/jaeger-operator/pkg/storage"
+	"github.com/jaegertracing/jaeger-operator/pkg/util"
+)
+
+// Ingester builds pods for jaegertracing/jaeger-ingester
+type Ingester struct {
+	jaeger *v1alpha1.Jaeger
+}
+
+// NewIngester builds a new Ingester struct based on the given spec
+func NewIngester(jaeger *v1alpha1.Jaeger) *Ingester {
+	if jaeger.Spec.Ingester.Size <= 0 {
```

Comment:
can we move this to normalize func?

Reply:
Possibly, but currently this is consistent with the other components. So should be handled in one go in another PR.